kasparovsyncd/sync/accepted_block.go

```diff
@@ -0,0 +1,52 @@
+package sync
+
+import (
+	"github.com/kaspanet/kaspad/rpcmodel"
+	"github.com/kaspanet/kasparov/database"
+	"github.com/kaspanet/kasparov/dbaccess"
+	"github.com/kaspanet/kasparov/dbmodels"
+
+	"github.com/pkg/errors"
+)
+
+func insertAcceptedBlocks(dbTx *database.TxContext, blocks []*rawAndVerboseBlock, blockHashesToIDs map[string]uint64) error {
+	acceptedBlocksToAdd := make([]interface{}, 0)
+	for _, block := range blocks {
+		dbAcceptedBlocks, err := makeAcceptedBlocks(blockHashesToIDs, block.Verbose)
+		if err != nil {
+			return err
+		}
+		for _, dbAcceptedBlock := range dbAcceptedBlocks {
+			acceptedBlocksToAdd = append(acceptedBlocksToAdd, dbAcceptedBlock)
+		}
+	}
+	err := dbaccess.BulkInsert(dbTx, acceptedBlocksToAdd)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func makeAcceptedBlocks(blockHashesToIDs map[string]uint64, verboseBlock *rpcmodel.GetBlockVerboseResult) ([]*dbmodels.AcceptedBlock, error) {
```

Comment:
This looks like it will be more appropriate if it were in the dbmodel package. 
However, right now the signature is very specific to the stuff you have available at insertAcceptedBlocks, which makes a bad fit to the more general dbmodel package.
Consider revising with a more general signature.

Also: I don't like the "make" prefix. Prefer something like "acceptedBlocksFromBlabla"

Reply:
Also, maybe after NOD-943 (Add acceptedBlockHashes to GetBlockVerboseResult) is merged you'll have all the info you need in verboseBlock.